Public/Src/Engine/Scheduler/PipRetryInfo.cs

```diff
@@ -0,0 +1,96 @@
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using BuildXL.Pips;
+using Logger = BuildXL.Scheduler.Tracing.Logger;
+using BuildXL.Utilities.Collections;
+using BuildXL.Utilities.Instrumentation.Common;
+using BuildXL.Utilities.Tracing;
+
+namespace BuildXL.Scheduler
+{
+    internal class PipRetryInfo

```

Comment:
This is no only for pip retry, but also for pip properties in general. That's why I suggested that we need two separate classes.

Reply:
Split out into a PipPropertyInfo class.